lib/Arch/Sleigh/PPCArch.cpp

```diff
@@ -0,0 +1,272 @@
+    REG(F2, fpr.f2.qword, f64);
+    REG(F3, fpr.f3.qword, f64);
+    REG(F4, fpr.f4.qword, f64);
+    REG(F5, fpr.f5.qword, f64);
+    REG(F6, fpr.f6.qword, f64);
+    REG(F7, fpr.f7.qword, f64);
+    REG(F8, fpr.f8.qword, f64);
+    REG(F9, fpr.f9.qword, f64);
+    REG(F10, fpr.f10.qword, f64);
+    REG(F11, fpr.f11.qword, f64);
+    REG(F12, fpr.f12.qword, f64);
+    REG(F13, fpr.f13.qword, f64);
+    REG(F14, fpr.f14.qword, f64);
+    REG(F15, fpr.f15.qword, f64);
+    REG(F16, fpr.f16.qword, f64);
+    REG(F17, fpr.f17.qword, f64);
+    REG(F18, fpr.f18.qword, f64);
+    REG(F19, fpr.f19.qword, f64);
+    REG(F20, fpr.f20.qword, f64);
+    REG(F21, fpr.f21.qword, f64);
+    REG(F22, fpr.f22.qword, f64);
+    REG(F23, fpr.f23.qword, f64);
+    REG(F24, fpr.f24.qword, f64);
+    REG(F25, fpr.f25.qword, f64);
+    REG(F26, fpr.f26.qword, f64);
+    REG(F27, fpr.f27.qword, f64);
+    REG(F28, fpr.f28.qword, f64);
+    REG(F29, fpr.f29.qword, f64);
+    REG(F30, fpr.f30.qword, f64);
+
+    REG(CRALL, iar.cr.qword, u64);
+    REG(CTR, iar.ctr.qword, u64);
+    REG(LR, iar.lr.qword, u64);
+    REG(XER, iar.xer.dword, u32);
+    REG(SPEFCR, iar.spefscr.qword, u64);
+    REG(ACC, iar.acc.qword, u64);
+
+    // These are actually bitflags within XER and CR respectively. These would
+    // normally be subregisters however, Sleigh treats these as entirely
+    // separate registers of size 1.
+    REG(XER_SO, xer_flags.so, u8);
+    REG(XER_OV, xer_flags.ov, u8);
+    REG(XER_CA, xer_flags.ca, u8);
+    REG(XER_COUNT, xer_flags.sl, u8);
+
+    REG(CR0, cr_flags.cr0, u8);
+    REG(CR1, cr_flags.cr1, u8);
+    REG(CR2, cr_flags.cr2, u8);
+    REG(CR3, cr_flags.cr3, u8);
+    REG(CR4, cr_flags.cr4, u8);
+    REG(CR5, cr_flags.cr5, u8);
+    REG(CR6, cr_flags.cr6, u8);
+    REG(CR7, cr_flags.cr7, u8);
+
+    REG(TBLR, tbr.tbl.qword, u64);
+    REG(TBUR, tbr.tbu.qword, u64);
+
+    REG(SPR103, sprg.r3.qword, u64);
+    REG(SPR104, sprg.r4.qword, u64);
+    REG(SPR105, sprg.r5.qword, u64);
+    REG(SPR106, sprg.r6.qword, u64);
+    REG(SPR107, sprg.r7.qword, u64);
+
+    REG(SPR203, l1cfg.r0.qword, u64);
+    REG(SPR204, l1cfg.r1.qword, u64);
+
+    REG(PC, pc, u64);
+
+    REG(TEA, signals.tea.qword, u64);
+  }
+
+  void
+  FinishLiftedFunctionInitialization(llvm::Module *module,
+                                     llvm::Function *bb_func) const override {
+    auto &context = module->getContext();
+    const auto u64 = llvm::Type::getInt64Ty(context);
+    const auto addr = u64;
+
+    auto &entry_block = bb_func->getEntryBlock();
+    llvm::IRBuilder<> ir(&entry_block);
+
+    const auto pc_arg = NthArgument(bb_func, kPCArgNum);
+    const auto state_ptr_arg = NthArgument(bb_func, kStatePointerArgNum);
+    ir.CreateStore(pc_arg,
+                   ir.CreateAlloca(addr, nullptr, kNextPCVariableName.data()));
+    ir.CreateStore(pc_arg, ir.CreateAlloca(addr, nullptr,
+                                           kIgnoreNextPCVariableName.data()));
```

Comment:
Can be polished as
```
auto mk_alloca = [ & ]( auto &from ) {
  return ir.CreateAlloca( addr, nullptr, from.data() );
};
ir.CreateStore( pc_arg, mk_alloca( kNextPCVariableName ) );
...
```

Reply:
[b50f203](https://github.com/lifting-bits/remill/pull/645/commits/b50f203b348ecb3efc044d836ee8634d04904ea5)